src/Core/Tools/Models/Business/ReferenceEvent.cs

```diff
@@ -245,10 +245,13 @@ public ReferenceEvent(ReferenceEventType type, IReferenceable source, ICurrentCo
     public string SignupInitiationPath { get; set; }
 
     /// <summary>
-    /// The will hold the details around the plan cusomer is and the plan they are migrating to. For example, "Teams Starter → Teams, Enterprise".
+    /// The upgrade applied to an account. The current plan is listed first,
+    /// followed by the plan they are migrating to. For example,
+    /// "Teams Starter → Teams, Enterprise".
     /// </summary>
     /// <value>
-    /// <see langword="null"/> when the event was not originated by an application.
+    /// <see langword="null"/> when the event was not originated by an application,
+    /// or when a downgrade occurred.
```

Comment:
⚠️ Now that the new and old product types are looked up separately, the field tracks downgrades, so this line isn't correct. It will be null, however, when either the new or old product description isn't tracked.

I see two ways to resolve this:

1. If this behavior is intended, update the docs.
2. If an upgrade should always be tracked, provide a fallback. Delete "or when a downgrade occurred" from the docs.

Here's one way the fallback could work. This falls back to the enum's name when there's no description.

```csharp
var oldDescription = _upgradePath.TryGetValue(oldProductType, out oldDescription) ? oldDescription : $"{oldProductType:G}";
```

Sorry for not recognizing this when I recommended the other code edit. 😓

Reply:
thank you so much, for pointing it out